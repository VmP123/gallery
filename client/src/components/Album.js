import React from 'react';
import { observer } from 'mobx-react'

@observer
class Album extends React.Component {
	render() {
		return (
			<div className="col s6 m4 xl3 album" onClick={this.props.openAlbum}>
				<img src={this.props.thumbnailUrl} className="responsive-img" alt=""/>
				<div className="info"><span>{this.props.name}</span></div>
			</div>
		);
	}
}

export default Album;
