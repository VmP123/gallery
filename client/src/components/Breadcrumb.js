import React from 'react';
import { observer } from 'mobx-react'
import { Link } from "react-router-dom";

@observer
class Breadcrumb extends React.Component {
	render() {
		return <Link to={this.props.url.substring(7)} className="breadcrumb"><span className="breadcrumb" >{this.props.name}</span></Link>
	}
}

export default Breadcrumb;
