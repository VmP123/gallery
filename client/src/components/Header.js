import React from 'react';

class Header extends React.Component {
	render() {
		return (
			<div className="header">
				Galleria
			</div>
		);
	}
}

export default Header;
