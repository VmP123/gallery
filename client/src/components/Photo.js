import React from 'react';
import { action } from 'mobx'
import { observer } from 'mobx-react'

@observer
class Photo extends React.Component {
	render() {
		return (
			<div className="photo col s6 m4 xl3" onClick={this.openPhoto}>
				<img className="responsive-img" src={this.props.thumbnailUrl} alt=""/>
			</div>
		);
	}

	@action openPhoto = () => {
		this.props.openPhoto(this.props.photoUrl);
	}
}

export default Photo;
