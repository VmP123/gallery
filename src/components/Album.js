import React from 'react';
import { observer } from 'mobx-react'

const Album = observer(
	class Album extends React.Component {
		render() {
			return (
				<div className="album">
					<img src={this.props.thumbnailUrl} alt=""/>
					<div className="info">
						<div>{this.props.name}</div>
						<div>
							<span>{this.props.photoCount}</span>
							{ this.props.albumCount > 0 ? <span>{this.props.albumCount}</span> : null}
						</div>
					</div>
				</div>
			);
		}
	}
)

export default Album;
