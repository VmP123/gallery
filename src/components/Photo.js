import React, { Component } from 'react';

export default class Photo extends Component {
    render() {
        return (
            <div className="photo">
                <img src={this.props.thumbnailUrl} alt=""/>
                <div className="info">{this.props.name}</div>
            </div>
        );
    }
}
