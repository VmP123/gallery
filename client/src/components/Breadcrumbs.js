import React from 'react';

import Breadcrumb from './Breadcrumb.js';

class Breadcrumbs extends React.Component {
	render() {
		return (
			<nav className="grey darken-3">
				<div className="nav-wrapper container row">
					<div className="col s12">
						{
							this.props.items.map((item, index) => {
								return <Breadcrumb key={index} getData={this.props.getData} name={item.name} url={item.url} />
							})
						}
					</div>
				</div>
			</nav>
		);
	}
}

export default Breadcrumbs;
