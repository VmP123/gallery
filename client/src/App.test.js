import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router'

import Breadcrumb from './components/Breadcrumb.js';
import Breadcrumbs from './components/Breadcrumbs.js';
import Header from './components/Header.js';
import Photo from './components/Photo.js';
import Album from './components/Album.js';

Enzyme.configure({ adapter: new Adapter() });

describe('Breadcrumb', () => {
	it('renders without crashing', () => {
		const tree = renderer
			.create(<MemoryRouter><Breadcrumb url={'http://www.google.com'} name={'Testing'} /></MemoryRouter>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});	
})

describe('Breadcrumbs', () => {
	it('renders without crashing', () => {
		const store = {
			breadcrumbs: [
				{
					name: 'Eka',
					url: 'http://www.google.com',
				},
				{
					name: 'Toka',
					url: 'http://www.google.com',
				}
			],
			getData() {
				console.log('Testing');
			}
		}
		
		const tree = renderer
			.create(<MemoryRouter><Breadcrumbs store={store} /></MemoryRouter>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});	
})

describe('Header', () => {
	it('renders without crashing', () => {		
		const tree = shallow(
			<Header />
		);
		expect(tree.debug()).toMatchSnapshot();
	});	
})

describe('Photo', () => {
	it('renders without crashing', () => {		
		const tree = shallow(
			<Photo thumbnailUrl={'http://www.google.com'} photoUrl={'http://www.google.com'} openPhoto={() => console.log('Testing')} />
		);
		expect(tree.debug()).toMatchSnapshot();
	});	
})

describe('Album', () => {
	it('renders without crashing', () => {		
		const tree = shallow(
			<MemoryRouter><Album name={'Testing'} thumbnailUrl={'http://www.google.com'} photoUrl={'http://www.google.com'} openAlbum={() => console.log('Testing')} /></MemoryRouter>
		);
		expect(tree.debug()).toMatchSnapshot();
	});	
})

