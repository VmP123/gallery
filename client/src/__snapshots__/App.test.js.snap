// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album renders without crashing 1`] = `
"<Router history={{...}}>
  <Album name=\\"Testing\\" thumbnailUrl=\\"http://www.google.com\\" photoUrl=\\"http://www.google.com\\" openAlbum={[Function: openAlbum]} />
</Router>"
`;

exports[`Breadcrumb renders without crashing 1`] = `
<a
  className="breadcrumb"
  href="/www.google.com"
  onClick={[Function]}
>
  <span
    className="breadcrumb"
  >
    Testing
  </span>
</a>
`;

exports[`Breadcrumbs renders without crashing 1`] = `
<nav
  className="grey darken-3"
>
  <div
    className="nav-wrapper container row"
  >
    <div
      className="col s12"
    >
      <a
        className="breadcrumb"
        href="/www.google.com"
        onClick={[Function]}
      >
        <span
          className="breadcrumb"
        >
          Eka
        </span>
      </a>
      <a
        className="breadcrumb"
        href="/www.google.com"
        onClick={[Function]}
      >
        <span
          className="breadcrumb"
        >
          Toka
        </span>
      </a>
    </div>
  </div>
</nav>
`;

exports[`Header renders without crashing 1`] = `
"<div>
  <nav className=\\"top-nav\\">
    <div className=\\"container\\">
      <div className=\\"nav-wrapper\\">
        <div>
          <div className=\\"col s12 m10 offset-m1\\">
            <h1 className=\\"grey-text text-darken-3 header-text\\">
              Valokuvagalleria
            </h1>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <inject-Breadcrumbs-with-store />
</div>"
`;

exports[`Photo renders without crashing 1`] = `
"<div className=\\"photo col s6 m4 xl3\\" onClick={[Function: res]}>
  <img className=\\"responsive-img\\" src=\\"http://www.google.com\\" alt=\\"\\" />
</div>"
`;
